### PR TITLE
implementation of the fb-person-datasource-mock

### PR DESCRIPTION
**Pull request by NortySpock**

### CODE REVIEW

**Comment by NortySpock** [comment]
Oh, ok, the Pull Request appears to have automatically updated itself when I pushed to master again. So then, these are my changes based on your suggestions. This excludes the PersonMetaData changes because I want to add those separately, as I believe this implementation works so far and I want to get the changes that resulted from your review checked over while I write all of the PersonMetaData code.

File: fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java [hunk 1]
```diff
@@ -3,50 +3,76 @@
...
+import java.util.UUID;
 
 /**
- * A Mock implementation of a {@link PersonDataSource} that holds persons in memory.
- * @author Jason W. Thompson - https://google.com/+JasonWThompson_SoftwareDeveloper
+ * A Mock implementation of a {@link PersonDataSource} that holds persons in
+ * memory.
+ * 
+ * @author Jason W. Thompson -
+ *         https://google.com/+JasonWThompson_SoftwareDeveloper
```
Comment (JasonWThompson) [inline fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java]: Add yourself as an author. It's optional if you put some sort of identifier (email/url) or not. You're welcome to if you want to.

File: fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java [hunk 2]
```diff
@@ -3,50 +3,76 @@
...
+    private static final Map<String, Person> PERSONS_BY_UID = new HashMap<>();
+    private static final Map<String, Person> DELETED_PERSONS_BY_UID = new HashMap<>();
+    private static final Map<String, Person> PERSON_METADATA_BY_UID = new HashMap<>();
+
     @Override
-    public void createPerson(Person person)
-    {
-        throw new UnsupportedOperationException("Not supported yet.");
+    public String createPerson(Person person) {
+        checkNotNull(person, "The parameter person cannot be null.");
```
Comment (JasonWThompson) [inline fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java]: A better message would be "The parameter person is null."

File: fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java [hunk 3]
```diff
@@ -3,50 +3,76 @@
...
+        return myUUID;
     }
 
     @Override
-    public void updatePerson(Person person)
-    {
-        throw new UnsupportedOperationException("Not supported yet.");
+    public void updatePerson(Person person) {
+        checkNotNull(person, "The parameter person cannot be null.");
+        PERSONS_BY_UID.put(person.getUID(), person);
```
Comment (JasonWThompson) [inline fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java]: You should also update the metadata too.

File: fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java [hunk 4]
```diff
@@ -3,50 +3,76 @@
...
-    {
-        throw new UnsupportedOperationException("Not supported yet.");
+    public String createPerson(Person person) {
+        checkNotNull(person, "The parameter person cannot be null.");
+        final String myUUID = UUID.randomUUID().toString();
+        SimplePerson myPerson = new SimplePerson();
+        myPerson.setUID(myUUID);
+        myPerson.setName(person.getName());
+
+        PERSONS_BY_UID.put(myUUID, myPerson);
```
Comment (JasonWThompson) [inline fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java]: Update the metadata collection here. Use the person UID as a key.

File: fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java [hunk 5]
```diff
@@ -3,50 +3,76 @@
...
+    public void updatePerson(Person person) {
+        checkNotNull(person, "The parameter person cannot be null.");
+        PERSONS_BY_UID.put(person.getUID(), person);
     }
 
     @Override
-    public void dropPersonByUID(String uid)
-    {
-        throw new UnsupportedOperationException("Not supported yet.");
+    public void dropPersonByUID(String uid) {
```
Comment (JasonWThompson) [inline fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java]: Be sure to remove the meta data too.

File: fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java [hunk 6]
```diff
@@ -3,50 +3,76 @@
...
+            PERSONS_BY_UID.remove(uid);
+        }
     }
 
     @Override
-    public void restorePersonByUID(String uid) throws UnsupportedOperationException
-    {
-        throw new UnsupportedOperationException("Not supported yet.");
+    public void restorePersonByUID(String uid) {
+        final Person temp = DELETED_PERSONS_BY_UID.get(uid);
```
Comment (JasonWThompson) [inline fb-person-datasource-mock/src/main/java/com/forcebook/person/datasource/mock/MockPersonDataSource.java]: Avoid variable names like `temp`. Something like `deletedPerson` would be more legible. If the method was longer and more complex, names like `temp` lose their meaning and can cause confusion.

**Comment by JasonWThompson** [comment]
The second round of code is much better so I went ahead and merged in the latest to master. There's still some minor issues and some stuff that needs to be implemented, but it's looking good so far. Go ahead and finish up what's left for the MockPersonDataSource and I'm going to try my best to get more code stubbed out to be worked on after I get back tonight.

**Pull request merged**